vaadin-radio-button-flow-demo/src/main/java/com/vaadin/flow/component/radiobutton/demo/RadioButtonGroupView.java

```diff
@@ -239,6 +247,91 @@ private void themeVariantsHorizontal() {
         addCard("Theme variants", "Direction", horizontal, vertical);
     }
 
+    private void dataViewRefreshItem() {
+        // begin-source-example
+        // source-example-heading: Refresh Items
+        RadioButtonGroup<Employee> radioButtonGroup = new RadioButtonGroup<>();
+        radioButtonGroup.setLabel("Assignable Employees: ");
+        Employee employee1 = new Employee("Employee One");
+        Employee employee2 = new Employee("Employee Two");
+        Employee employee3 = new Employee("Employee Three");
+        RadioButtonGroupListDataView<Employee> dataView = radioButtonGroup
+                .setItems(employee1, employee2, employee3);
+        radioButtonGroup.setValue(employee3);
+        radioButtonGroup.addThemeVariants(RadioGroupVariant.LUMO_VERTICAL);
+
+        Button updateButton = new Button("Update second employee's name",
+                click -> {
+                    employee2.setTitle("Employee 2");
+                    dataView.refreshItem(employee2);
+                });
+        // end-source-example
+
+        addCard(DATA_VIEW, "Refresh Items", radioButtonGroup, updateButton);
+    }
+
+    private void dataViewAddItem() {
+        // begin-source-example
+        // source-example-heading: Add Item
+        RadioButtonGroup<Employee> radioButtonGroup = new RadioButtonGroup<>();
+        radioButtonGroup.setLabel("Assignable Employees: ");
+        RadioButtonGroupListDataView<Employee> dataView = radioButtonGroup
+                .setItems(getEmployeeList());
+        radioButtonGroup.addThemeVariants(RadioGroupVariant.LUMO_VERTICAL);
+        AtomicInteger employeeCounter = new AtomicInteger(1);
+        Button addButton = new Button("Add to Options",
+                click -> dataView.addItem(new Employee(
+                        "Employee " + (employeeCounter.incrementAndGet()))));
+        Button removeButton = new Button("Remove from Options", click -> {
+            if (dataView.getItemCount() > 0)
```

Comment:
Extract `dataView.getItemCount()` to an extra variable to not call it twice. Add curly braces to if clause